pkg/apis/jaegertracing/v1/options_test.go

```diff
@@ -148,3 +150,38 @@ func TestUpdate(t *testing.T) {
 	// verify
 	assert.Equal(t, o.opts["key"], "new")
 }
+
+func TestStringMap(t *testing.T) {
+	o := NewOptions(nil)
+	err := o.UnmarshalJSON([]byte(`{"firstsarg":"v1", "additional-headers":["whatever:thing", "access-control-allow-origin:blerg"]}`))
```

Comment:
I'll add some real examples, 

This is testing a new method I introduced in this PR. `StringMap`, this method only returns a map with strings (mimic the old behavior when we didn't support arrays) , it omits repeated arguments, I did in this way because this map is used in a lot of places, and I wand to preserve that behaviour for those places which is used, also I don't want to do type asserting each time the option map is used.